test/jdk/javax/swing/JTable/PrintAllPagesTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/*
+ * @test
+ * @bug 8257810
+ * @summary  Verifies if all pages are printed if scrollRectToVisible is set.
+ * @run main/manual PrintAllPagesTest
+ */
+import java.awt.BorderLayout;
+import java.awt.FlowLayout;
+import java.awt.print.PrinterException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+import javax.swing.table.AbstractTableModel;
+import javax.swing.table.TableModel;
+import javax.swing.WindowConstants;
+
+public class PrintAllPagesTest {
+    static JFrame f;
+    static JDialog dialog;
+    static JTable table;
+    static volatile boolean testResult = false;
+    final static CountDownLatch latch = new CountDownLatch(1);
+    static  boolean ret = false;
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            SwingUtilities.invokeAndWait(() -> {
+                createUI();
+                printAllPagesTest();
+            });
+
+            Thread.sleep(1000);
+            SwingUtilities.invokeAndWait(() -> {
+                try {
+                    ret = table.print();
+                } catch (PrinterException e) {
+                    throw new RuntimeException("Printing failed: " + e);
+                }
+            });
+
+            if (!testResult) {
+                throw new RuntimeException("Only 1st page is printed out of multiple pages");
+            }
```

Comment:
Now the test always fails whether printing is called or not.

I handled this situation in the patch I proposed.